drivers/adc/ad463x/iio_ad463x.c

```diff
@@ -91,31 +151,184 @@ static int32_t _iio_ad463x_prepare_transfer(struct iio_ad463x *desc,
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ * @return ret    - Result of the reading procedure.
+ *		    In case of success, the size of the read data is returned.
+*/
+static int ad463x_iio_read_raw(void *dev, char *buf, uint32_t len,
+			       const struct iio_ch_info *channel, intptr_t priv)
+{
+	uint32_t temp[20];
+	int ret;
+	int32_t temp2;
+	struct iio_ad463x *iio_desc = dev;
+
+	/* Exit register configuration mode */
+	ret = ad463x_exit_reg_cfg_mode(iio_desc->ad463x_desc);
+
+	ret = ad463x_read_data(iio_desc->ad463x_desc, temp, 2);
+	if (ret < 0)
+		return ret;
+	/** Get sample in the middle according to the channel */
+	if(channel->ch_num == 0)
+		temp2 = temp[0];
+	else
+		temp2 = temp[1];
+	temp2 = no_os_sign_extend32(temp2, REAL_BITS - 1);
+	
+	ad463x_enter_config_mode(iio_desc->ad463x_desc);
+
+	return iio_format_value(buf, len, IIO_VAL_INT, 1, &temp2);
+}
+
+/**
+ * @brief Handles the read request for Scale attribute.
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ * @return ret    - Result of the reading procedure.
+ *		    In case of success, the size of the read data is returned.
+*/
+static int ad463x_iio_read_scale(void *dev, char *buf, uint32_t len,
+				 const struct iio_ch_info *channel, intptr_t priv)
+{
+	struct iio_ad463x *iio_desc = dev;
+	struct ad463x_dev *ad463x_dev;
+	int32_t vals[2];
+
+	if (!iio_desc)
+		return -EINVAL;
+
+	ad463x_dev = iio_desc->ad463x_desc;
+	if(ad463x_dev->has_pgia) {
+		vals[0] = ad463x_dev->scale_table[ad463x_dev->pgia_idx][0];
+		vals[1] = ad463x_dev->scale_table[ad463x_dev->pgia_idx][1];
+		return iio_format_value(buf, len, IIO_VAL_INT_PLUS_NANO, 2, vals);
+	}
+
+	vals[0] = (ad463x_dev->vref * 2) / 1000;
+	vals[1] = ad463x_dev->capture_data_width;
+	return iio_format_value(buf, len, IIO_VAL_FRACTIONAL_LOG2, 2, vals);
+}
+
+/**
+ * @brief Handles the write request for scale attribute.
+ * @param dev     - The iio device structure.
+ * @param buf	  - Command buffer to be filled with requested data.
+ * @param len     - Length of the received command buffer in bytes.
+ * @param channel - Command channel info.
+ * @param priv    - Command attribute id.
+ * @return ret    - Result of the reading procedure.
+ *		    In case of success, the size of the read data is returned.
+*/
+static int ad463x_iio_store_scale(void *dev, char *buf, uint32_t len,
+				  const struct iio_ch_info *channel, intptr_t priv)
+{
+	struct iio_ad463x *iio_desc = dev;
+	struct ad463x_dev *ad463x_dev;
+	ad463x_dev = iio_desc->ad463x_desc;
+
+	int32_t vals[2];
+	/** Scale only configurable if PGIA is available */
+	if(ad463x_dev->has_pgia) {
```

Comment:
We usually use negative logic in such situations.

```
if (!ad463x_dev->has_pgio)
    return 0;
...
```

Reply:
Noted.